src/targets/gpu/compile_pointwise.cpp

```diff
@@ -57,7 +59,17 @@ operation compile_pointwise(context&,
                                    {"preamble", preamble}});
     return compile_hip_code_object(src, options);
 }
-
+// Overload of compile_pointwise without global, local calculates global value at runtime
+operation compile_pointwise(context& ctx,
+                            const std::vector<shape>& inputs,
+                            const std::string& lambda,
+                            const std::string& preamble)
+{
+    size_t local(1024);
+    size_t n      = inputs.size() > 0 ? inputs[0].elements() : 1024;
```

Comment:
The inputs will not ever be empty, we can probably just add an assert for that, and just use `inputs.front().elements()`.

Reply:
done.  inputs.first().elements()